pkg/appmanager/appManager.go

```diff
@@ -1199,6 +1221,31 @@ func (appMgr *Manager) setServerSslProfile(
 	rsCfg *ResourceConfig,
 	route *routeapi.Route,
 ) {
+	// First handle the Default for SNI profile
+	if appMgr.routeConfig.ServerSsl != "" {
+		// User has provided a name
+		profile := ProfileRef{
+			Name:      appMgr.routeConfig.ServerSsl,
+			Partition: rsCfg.Virtual.Partition,
+			Context:   customProfileServer,
+		}
+		rsCfg.Virtual.AddOrUpdateProfile(profile)
+	} else {
+		// No provided name, so we create a default
+		profile := ProfileRef{
+			Name:      "default-route-serverssl",
+			Partition: rsCfg.Virtual.Partition,
+			Context:   customProfileServer,
+		}
+		// This is just a basic profile, so we don't need all the fields
+		cp := NewCustomProfile(profile, "", "", "", true, appMgr.customProfiles)
```

Comment:
If the user doesn't configure a client or server ssl default are we creating a new default profile every time we set the profiles - even if we've already created one. Shouldn't this happen for just the first set? Then we can check the profs each time through?

Reply:
This does create a new profile every time, but it is always the same, and overrides the previous entry that stored the same profile. Nothing different is written out, but we are calling this piece of code every time.

A similar process happens later in this function when processing the Route profile. I can add a check here to only create the new profile if it doesn't already exist.

Reply:
Spoke offline, can do exists check from map lookup.